From 47898ac61eef57bef92cd8dee95f76cb2f1071ff Mon Sep 17 00:00:00 2001
From: Colin Patrick Mccabe <cmccabe@cloudera.com>
Date: Mon, 14 Mar 2016 14:59:11 -0700
Subject: [PATCH 1317/2848] HDFS-9947. Block#toString should not output
 information from derived classes (cmccabe)

(cherry picked from commit 9a43094e12ab8d35d49ceda2e2c5f83093bb3a5b)
(cherry picked from commit 8ac3fc7540eb7006534cadc6755e7196e764b0ec)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java

Change-Id: Ia364fb80cdbfc0357a58957dfc8fe6a07cf64a8a
---
 .../org/apache/hadoop/hdfs/protocol/Block.java     |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java
index f86854d..0b0c42a 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java
@@ -158,9 +158,9 @@ public void setGenerationStamp(long stamp) {
    */
   @Override
   public String toString() {
-    StringBuilder sb = new StringBuilder();
-    this.appendStringTo(sb);
-    return sb.toString();
+    sb.append(BLOCK_FILE_PREFIX).
+       append(b.blockId).append("_").
+       append(b.generationStamp);
   }
 
   public void appendStringTo(StringBuilder sb) {
-- 
1.7.9.5

